### PR TITLE
[BPF] extra conntrack counters and stats

### PR DESCRIPTION
**Pull request by tomastigera**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: additional conntrack counters
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by ioworker0** [comment]
Nice work ~

It seems that we added stats for conntrack in the second commit, which is really useful. Would it be possible to also add counters for conntrack creation failure events?

**Comment by tomastigera** [comment]
> Would it be possible to also add counters for conntrack creation failure events?

That is added in the first commit `deny_reason(ctx, CALI_REASON_CT_CREATE_FAILED);` and printed in `dump counters`

File: node/Makefile [hunk 1]
```diff
@@ -265,12 +265,12 @@ sub-image-fips-%:
...
-	$(DOCKER_BUILD) --build-arg BIN_DIR=$(NODE_CONTAINER_BIN_DIR) --build-arg BIRD_IMAGE=$(BIRD_IMAGE) --build-arg GIT_VERSION=$(GIT_VERSION) -t $(NODE_IMAGE):latest-$(ARCH) -f ./Dockerfile.$(ARCH) .
+	$(DOCKER_BUILD) --network=host --build-arg BIN_DIR=$(NODE_CONTAINER_BIN_DIR) --build-arg BIRD_IMAGE=$(BIRD_IMAGE) --build-arg GIT_VERSION=$(GIT_VERSION) -t $(NODE_IMAGE):latest-$(ARCH) -f ./Dockerfile.$(ARCH) .
 	$(MAKE) retag-build-images-with-registries VALIDARCHES=$(ARCH) IMAGETAG=latest
 	touch $@
 
 $(NODE_CONTAINER_FIPS_CREATED): $(REMOTE_DEPS) ./Dockerfile.$(ARCH) $(NODE_CONTAINER_BINARY) $(INCLUDED_SOURCE) $(NODE_CONTAINER_FILES) $(TOOLS_MOUNTNS_BINARY)
-	$(DOCKER_BUILD) --build-arg BIN_DIR=$(NODE_CONTAINER_BIN_DIR) --build-arg BIRD_IMAGE=$(BIRD_IMAGE) --build-arg GIT_VERSION=$(GIT_VERSION) -t $(NODE_IMAGE):latest-fips-$(ARCH) -f ./Dockerfile.$(ARCH) .
+	$(DOCKER_BUILD) --network=host --build-arg BIN_DIR=$(NODE_CONTAINER_BIN_DIR) --build-arg BIRD_IMAGE=$(BIRD_IMAGE) --build-arg GIT_VERSION=$(GIT_VERSION) -t $(NODE_IMAGE):latest-fips-$(ARCH) -f ./Dockerfile.$(ARCH) .
```
Comment (sridhartigera) [inline node/Makefile:273]: Is the ```--network=host``` change intended?

File: felix/bpf/counters/counters.go [hunk 2]
```diff
@@ -155,6 +158,18 @@ var descriptions DescList = DescList{
...
+		Counter:  SourceCollision,
+		Category: "Other", Caption: "packets hitting NAT source collision",
+	},
+	{
+		Counter:  ConntrackCreateFailed,
+		Category: "Dropped", Caption: "failed to create conntrack",
+	},
+	{
+		Counter:  SourceCollisionFailed,
+		Category: "Dropped", Caption: "packets hitting NAT source collision failed",
```
Comment (sridhartigera) [inline felix/bpf/counters/counters.go:171]: Can the Caption for SourceCollision and SourceCollisionFailed be better to capture the difference between these 2? Its a bit confusing. Maybe something like `packets hitting NAT source collision unresolved`.

File: node/Makefile (hunk 1 above)
Comment (tomastigera) [reply node/Makefile:273]: not 100% intended, but would it hurt?

Comment (tomastigera) [reply node/Makefile:273]: other builds in that file also use host, and this is going to make my life easier, so I just want to smuggle it in

**Comment by ioworker0** [comment]
> > Would it be possible to also add counters for conntrack creation failure events?
> 
> That is added in the first commit `deny_reason(ctx, CALI_REASON_CT_CREATE_FAILED);` and printed in `dump counters`

Ah, I clearly missed that.

**Review by fasaxc — APPROVED** [review]
LGTM

**Pull request merged**